src/framework/preferences/css/stylus/utils/Themes.styl

```diff
@@ -1,250 +1,376 @@
-        fColor = contrastThemes[themeName].foregroundColor;
-        bColor = contrastThemes[themeName].backgroundColor;
+.fl-theme-yb {
+    --fl-fgColor: #ffff00;
+    --fl-bgColor: #000000;
+}
 
-        {themeName} {
-            .fl-prefsEditor {
+.fl-theme-by {
+    --fl-fgColor: #000000;
+    --fl-bgColor: #ffff00;
+}
 
-                // ON/OFF switch
-                .fl-switchUI-control:focus {
-                    box-shadow: 0 0 0 0.2em bColor,
-                                0 0 0 0.35em fColor;
-                }
+.fl-theme-lgdg {
+    --fl-fgColor: #bdbdbb;
+    --fl-bgColor: #555555;
+}
 
-                .fl-switchUI-controlKnob {
-                    background-color: fColor !important;
-                }
-            }
-        }
+.fl-theme-gd {
+    --fl-fgColor: #888888;
+    --fl-bgColor: #222222;
+    --fl-linkColor: #8080ff;
+    --fl-disabledColor: #cc0000;
+    --fl-selectedFgColor: #ffffff;
+    --fl-selectedBgColor: #008000;
+    --fl-buttonFgColor: #ffffff;
+    --fl-buttonBgColor: #000000;
+}
 
-        .fl-prefsEditor .fl-choice label {
-            &{themeName},
-            &{themeName} span,
-            &{themeName} .fl-crossout {
-                background-color: bColor !important;
-                border-color: fColor !important;
-                color: fColor !important;
-            }
-        }
-    }
+.fl-theme-gw {
+    --fl-fgColor: #6c6c6c;
+    --fl-bgColor: #ffffff;
+    --fl-linkColor: #6666cc;
+    --fl-disabledColor: #993333;
+    --fl-selectedFgColor: #ffffff;
+    --fl-selectedBgColor: #404080;
+    --fl-buttonFgColor: #000000;
+    --fl-buttonBgColor: #ffffff;
 }
 
-// For the separated panel
-get-colors-for-separatedPanelPrefsEditor(fColor, bColor) {
-    background-color: bColor;
-    border-color: fColor;
-    color: fColor;
+.fl-theme-bbr {
+    --fl-fgColor: #000000;
+    --fl-bgColor: #bb9966;
+    --fl-linkColor: #000099;
+    --fl-disabledColor: #600000;
+    --fl-selectedFgColor: #ffffff;
+    --fl-selectedBgColor: #000070;
+    --fl-buttonFgColor: #000000;
+    --fl-buttonBgColor: #ffd18a;
 }
 
-build-themes-separatedPanelPrefsEditor(contrastThemes) {
-    for themeName in contrastThemes {
-        fColor = contrastThemes[themeName].foregroundColor;
-        bColor = contrastThemes[themeName].backgroundColor;
-
-        {themeName} {
-            background-image: none !important;
-            get-colors-for-separatedPanelPrefsEditor(fColor, bColor);
-
-            .fl-prefsEditor-separatedPanel {
-                .fl-panelBar, .fl-prefsEditor-reset {
-                    get-colors-for-separatedPanelPrefsEditor(fColor, bColor);
-                }
-                .fl-prefsEditor-buttons { //FLUID-4991: The shadow effect around the hide/show preferences bug fix.
-                    box-shadow:none;
-                }
-            }
+// For the general theme styling that applies to html elements and infusion components
+build-themes-Enactors() {
```

Comment:
It doesn't appear that this actually needs to be a function any more. If it is included as function just to provide a way to import the colour variables independently, we should split them into a separate file. If the reason is about importing this particular bit of styling to a specific location within the enactors file, I believe that Stylus files can be imported into any location, only CSS files have to be imported at the top.

Reply:
Speaking of the Enactors.styl file, we actually may want to provide options for using the styling for enactors independently. That is, there may be cases where the themes aren't used, but the rest of the enactors are. We don't want to have to include all those styles unnecessarily. Similar for the other preferences. I've written up [FLUID-6552](https://issues.fluidproject.org/browse/FLUID-6552) for that.

Reply:
Thanks, @jobara. Resolved your first comment in a1b6fba.